### PR TITLE
Address creation

### LINKED ISSUE #344: Room cannot be added
The creating a room in the frontend does not work because no address is set.

### PR DESCRIPTION
**Pull request by lukasjuhrich**
This PR
- Introduces typescript
- Creates a generic `typeahead` component
- Fixes #344 by introducing `get_or_create_address` and proper form inputs for the `create_room` endpoint.

The room creation has two key features:
- All fields except for `addition` are pre-filled by these values that have the highest occurrence among the addresses of the other rooms in the respective buildings
- There is an simple autocompletion for each field.

### CODE REVIEW

File: web/resources/js/hello.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import $ from "jquery";
```
Comment (georgeto) [inline web/resources/js/hello.ts:1]: Remove this commit from the PR? With `generic-typeahead.ts` there seems to be a proper typescript module now.

File: web/resources/js/hello.ts [hunk 2]
```diff
@@ -5,3 +5,14 @@ document.addEventListener('DOMContentLoaded', () => {console.log("This is runnin
 $(function () {
     $('[data-toggle="tooltip"]').tooltip()
 });
+
```
Comment (georgeto) [inline web/resources/js/hello.ts:8]: Remove this commit from the PR? With `generic-typeahead.ts` there seems to be a proper typescript module now.

**Comment by ibot3** [comment]
For me, the `test_address_unique` test is failing. Can you confirm that?

**Comment by lukasjuhrich** [comment]
> For me, the `test_address_unique` test is failing. Can you confirm that?

Yes. Running the complete suite again and fixing tests afterwards.

**Comment by lukasjuhrich** [comment]
Fixed, thanks

**Pull request merged**